sdks/python/apache_beam/runners/direct/evaluation_context.py

```diff
@@ -316,16 +320,22 @@ class DirectUnmergedState(InMemoryUnmergedState):
   def __init__(self):
     super(DirectUnmergedState, self).__init__(defensive_copy=False)
 
+  # TODO(mariagh): make a selective deepcopy of just what is needed
+  # to preserve the state while a bundle is processed.
+  def clone(self):
+    return copy.deepcopy(self)
+
 
 class DirectStepContext(object):
   """Context for the currently-executing step."""
 
   def __init__(self, keyed_existing_state):
     self.keyed_existing_state = keyed_existing_state
+    self.partial_keyed_state = {}
```

Comment:
<!--new_thread; commit:6ce5f121f4998fe51aa9ac13064dce692cd1228b; resolved:0-->
Can you please add a comment on how this is keeping track of the changed state?